pkg/reporting/batch_jsonarray.go

```diff
@@ -51,7 +51,7 @@ func NewJSONArrayBuilder[E any](buf IOBuffer, nestedFields ...string) *JSONArray
 
 func (b *JSONArrayBuilder[E]) Add(event E) {
 	if b.started {
-		if _, err := b.buf.Write([]byte("\n\t,")); err != nil {
+		if _, err := b.buf.Write([]byte(",")); err != nil {
```

Comment:
nit: should we have a unittest for that or e2e, may be not now (to speed up fixing) but later?